internal/pkg/service/appsproxy/dns/transport.go

```diff
@@ -0,0 +1,66 @@
+package dns
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	"golang.org/x/net/http2"
+)
+
+// TLSHandshakeTimeout specifies the default timeout of TLS handshake.
+const TLSHandshakeTimeout = 5 * time.Second
+
+// ResponseHeaderTimeout specifies the default amount of time to wait for a server's response headers.
+const ResponseHeaderTimeout = 15 * time.Second
+
+// MaxIdleConnections specifies the default maximum number of open connections at all.
+const MaxIdleConnections = 128
+
+// MaxConnectionsPerHost specifies the default maximum number of open connections to a host.
+const MaxConnectionsPerHost = 32
+
+// HTTP2ReadIdleTimeout is the timeout after which a health check using ping frame will be carried out.
+const HTTP2ReadIdleTimeout = 2 * time.Second
+
+// HTTP2PingTimeout is the timeout after which the connection will be closed  if a response to Ping is not received.
+const HTTP2PingTimeout = 2 * time.Second
+
+// HTTP2WriteByteTimeout is the timeout after which the connection will be closed no data can be written to it.
+const HTTP2WriteByteTimeout = 15 * time.Second
+
+// NewDNSSkippingHTTPTransport creates an HTTP transport which can skip DNS resolution if the address is already within request context.
+// Timeouts are optimized for our usage in reverse proxy.
+func NewDNSSkippingHTTPTransport(addressCtxKey any) (*http.Transport, error) {
+	dialer := newDialer()
+
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+
+	transport.DialContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+		address, ok := ctx.Value(addressCtxKey).(string)
```

Comment:
`dns/transport.go` creates `http.Transport` :thinking: 

The `dns` package should contain only the creation of the `Dialer`, there the DNS layer ends.

HTTP timeouts are not related to the DNS.

Do we have a better place where we would move the transport creation?

Reply:
> I want to split the work into two parts. 

I see the following separate parts there:
- Use the custom `http.Transport` for the proxy upstream (discussed on Slack).
- Rendering error pages : rendering the "spinner" page.

> Then second is here where the DNS resolution is skipped if the resolved address already exists in the request context.
- I don't understand it, can you please explain? Or let's call shortly.

Reply:
Ok, if you call this a `manual lookup` then it fits. I take it that the  `http.Transport` to do the lookup, so you don't do any manual lookup, you just set the Transport.

>  I'm doing is moving the manual lookup elsewhere because I can't render the reloading page from inside the transport.
So this is probably the problematic part. You can't and shouldn't do that in the Transport.

You should implement a custom PageWriter and catch the error there (see Slack discussion).

Reply:
See :two: https://keboolaglobal.slack.com/archives/C055WKPHYSD/p1711014587799899?thread_ts=1711008295.446849&cid=C055WKPHYSD